lib/api/unversioned/types.go

```diff
@@ -23,6 +23,7 @@ type Resource interface {
 type ResourceObject interface {
 	Resource
 	GetResourceMetadata() ResourceMetadata
+	String() string
```

Comment:
Given the very specific behavior of this String() method, I think it is worth adding a doc comment.